mongodb/docs-worker-pool
# DOP-3818: Create webhook to mark build data for deletion

### Ticket

DOP-3818

### Testing

Assuming that this branch is in pre-production, you can do the following to test:
1. Ensure you have the `stg` GitHub build trigger webhook set up in a fork of a docs repo.
2. Create and push a new branch to **your fork**. Content changes are optional. This should trigger an Autobuilder job.
3. Follow the steps in this [wiki page](https://wiki.corp.mongodb.com/display/DE/Set+Up+a+GitHub+Repo+to+Mark+Build+Data+for+Deletion) to add the new deletion webhook in **your fork** of the docs repo. Since this is a test, this can be done on a fork and not an actual upstream (mongodb/10gen) repo.
4. Create a PR for the new branch. Make sure the base branch is **your fork** where you have the new deletion webhook added.
5. Once the Autobuilder job is complete, close or merge the new branch. This will trigger the new deletion webhook to mark data for deletion.
6. Ensure that the metadata and updated page ASTs associated with the project + branch you created have a `deleted` field set to true.

**Screenshots:**
<img width="1143" alt="Screen Shot 2023-07-21 at 1 17 02 PM" src="https://github.com/mongodb/docs-worker-pool/assets/27821750/425221fd-f766-43ab-9880-5ff225b45d77">
<img width="1150" alt="Screen Shot 2023-07-21 at 1 16 20 PM" src="https://github.com/mongodb/docs-worker-pool/assets/27821750/9a5633ff-af8c-4ea6-99ba-623ab4df2da9">

### Notes

* Adds a new webhook to mark build data for deletion. The hope is to have the Gatsby source plugin delete its internal nodes pertaining to data where the `deleted` field is set to true. This will hopefully help reduce the memory usage for Gatsby Cloud sites in a more immediate term. The data in the collection will actually be deleted by the existing TTLs when the time comes.
* The new webhook/function has only be added in v1 right now. Some of the env setup has already been done in the CDK code just to have the values there, but adding the webhook/function to the v2 paths in a future separate PR might be more suitable to help keep the scope of changes and testing down.
